File: validation/connectors/validateConnectors.ts
import { extractPostgresRuntime, extractRedisRuntime, runLiveRuntime } from "../../src/connectors/index.js";

const pg = extractPostgresRuntime({ tables: ["users"], schemas: ["public"] });
const redis = extractRedisRuntime({ keys: ["a", "b"] });
const live = runLiveRuntime({ database_type: "postgresql" });

const results = {
  postgres: pg.database_type === "postgresql",
  redis: redis.database_type === "redis",
  live: live.bounded === true,
};

console.log("PASS", results);
if (!Object.values(results).every(Boolean)) process.exit(1);

File: validation/connectors/validate_connectors.dart
import 'package:webweavex/webweavex.dart';

void main() {
  final pg = extractPostgresRuntime({'tables': ['t']});
  final live = runLiveRuntime();
  if (pg['database_type'] != 'postgresql' || live['bounded'] != true) {
    throw StateError('connector validation failed');
  }
  print('PASS connectors');
}
